### PR TITLE
set up an initial development environment

### PR DESCRIPTION
**Pull request by purplepig4657**

### CODE REVIEW

**Comment by hygoni** [comment]
Do not create a separate commit just for code formatting.

You need to follow the coding style in your initial commit.
I suggest running the code formatting script before committing.
Alternatively, you can run it after committing, but make sure to amend the changes into the original commit.

File: RuntimeLibrary/plsan_handler.h [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#ifndef PLSAN_HANDLER_H
+#define PLSAN_HANDLER_H
+
+#include "tuple"
+
+namespace __plsan {
+
+enum AddrType { NonDynAlloc, DynAlloc };
+enum ExceptionType { None, RefCountZero, DynAllocMemNonFree };
```
Comment (hygoni) [inline RuntimeLibrary/plsan_handler.h:9]: RefCountZero and DynAllocMemNonFree are the same thing. 
Except that, anything else LGTM

Comment (hygoni) [reply RuntimeLibrary/plsan_handler.h:9]: @purplepig4657 
Oh, but Please ensure that it remains possible to build
 We don't want to break what worked before

**Review by hygoni — CHANGES_REQUESTED** [review]

File: RuntimeLibrary/plsan_shadow.cpp [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+namespace __plsan {
+
+PlsanShadow::PlsanShadow() {
+  shadow_addr = mmap(NULL, MMAP_SIZE, PROT_READ | PROT_WRITE,
+                     MAP_ANONYMOUS | MAP_PRIVATE | MAP_NORESERVE, -1, 0);
+  if (shadow_addr == MAP_FAILED)
+    throw "mmap failed\n";
+}
+
```
Comment (kimseongyu) [inline RuntimeLibrary/plsan_shadow.cpp:13]: when program is over, we have to deallocate shadow memory.
So, add a Destructor. Like this
```
PlsanShadow::~PlsanShaow(){
    munmap(shadow_addr, MMAP_SIZE);
}
```

File: RuntimeLibrary/plsan.cpp [hunk 3]
```diff
@@ -3,10 +3,13 @@
...
  */
 
+#include "plsan.h"
+
 #include <cstddef>
 
 /* Initialization routines called before main() */
 __attribute__((constructor)) void __plsan_init() { /* TODO: */
+  __plsan::Plsan *plsan;
 }
```
Comment (kimseongyu) [inline RuntimeLibrary/plsan.cpp:14]: when program is over, if we have to do something use this.
exmaple: deallocate shadow memory

```
/* finialization routines called after main()*/
__attribute__((destructor)) void __plsan_fini()
{
    delete plsan;
}
```

File: RuntimeLibrary/plsan_handler.h (hunk 1 above)
Comment (purplepig4657) [reply RuntimeLibrary/plsan_handler.h]: @hygoni 
Please check changed CMakeLists.txt

**Review by hygoni — APPROVED** [review]

**Pull request merged**